client/src/JobPaginator.jsx

```diff
@@ -1,326 +1,69 @@
-                    if (!window.confirm(msg)) {
-                        return;
-                    }
-                }
-                this.ajax.ajax({
-                    type: "PUT",
-                    url: this.ajax.getJobUrl(this.state.category, this.state.job, ""),
-                    contentType: "application/json",
-                    data: JSON.stringify({ order: this.state.order, published: publish }),
-                    success: function () {
-                        alert("Success!");
-                        window.location.assign("/paginate"); // TODO: Route better?
-                        dispatchEvent(new Event("Prep.saved"));
-                    },
-                    error: function () {
-                        alert("Unable to save!");
-                    },
-                });
-            }.bind(this)
-        );
-    }
-
-    toggleZoom() {
-        this.newState = this.state;
-        this.newState.zoom = !this.newState.zoom;
-        this.setState(this.newState);
-    }
-
-    componentDidMount() {
-        this.loadJob(this.props.category, this.props.job);
-    }
-
-    render() {
-        var preview = this.state.zoom ? (
-            <PaginatorZoomy img={this.getImageUrl(this.state.currentPage, "large")} />
-        ) : (
-            <PaginatorPreview img={this.getImageUrl(this.state.currentPage, "medium")} />
-        );
-        return (
-            <div className={this.state.active ? "" : "hidden"} id="paginator">
-                <div className="row">
-                    <div className="six col">{preview}</div>
-                    <div className="six col">
-                        <p>
-                            {this.state.category} &gt; {this.state.job}
-                        </p>
-                        <PaginatorControls paginator={this} />
-                        <PaginatorList paginator={this} pageCount={this.state.order.length} />
-                    </div>
+import useJobPaginator from "./useJobPaginator";
+
+const JobPaginator = ({ initialCategory, initialJob }) => {
+    const {
+        state: { active, currentPage, zoom, order, category, job },
+        action: {
+            setLabel,
+            getLabel,
+            getMagicLabel,
+            setPage,
+            nextPage,
+            prevPage,
+            deletePage,
+            save,
+            useEffect,
+            autonumberFollowingPages,
+            getImageUrl,
+            setZoom,
+        },
+    } = useJobPaginator(initialCategory, initialJob);
+
+    useEffect();
+
+    return (
+        <div className={active ? "" : "hidden"} id="paginator">
+            <div className="row">
+                <JobPaginatorZoom zoom={zoom} getImageUrl={(size) => getImageUrl(order[currentPage], size)} />
+                <div className="six col">
+                    <p>
+                        {category} &gt; {job}
+                    </p>
+                    <PaginatorControls
```

Comment:
This is still overly wordy for me. I'm planning to move some states and functions to a shared state, since they are redundantly used in the child components anyway.

Reply:
Yes, that makes sense, though if you want to keep this as-is for now and work on that in a separate iteration, that would be fine with me. Might be good to get the current working state merged before making further revisions.